tests/components/homee/test_cover.py

```diff
@@ -0,0 +1,93 @@
+"""Test homee covers."""
+
+from unittest.mock import MagicMock
+
+from pyHomee import HomeeNode
+
+from homeassistant.components.homee.const import DOMAIN
+from homeassistant.components.homee.cover import HomeeCover
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry, load_json_object_fixture
+
+
+async def test_cover_open(
+    hass: HomeAssistant, mock_homee: MagicMock, mock_config_entry: MockConfigEntry
+) -> None:
+    """Test an open cover."""
+    mock_config_entry.add_to_hass(hass)
+    mock_config_entry.runtime_data = mock_homee
+
+    # Cover open, tilt open.
+    cover_json = load_json_object_fixture("cover1.json", DOMAIN)
+    cover_node = HomeeNode(cover_json)
+    cover1 = HomeeCover(cover_node, mock_config_entry)
+
+    assert cover1.unique_id == "00055511EECC-3-101"
+
+    assert cover1.state == "open"
+    assert cover1.is_closed is False
+    assert cover1.is_closing is False
+    assert cover1.is_opening is False
+    assert round(cover1.current_cover_position) == 100
+    assert round(cover1.current_cover_tilt_position) == 100
```

Comment:
Okay so we don't want to touch internals in tests. Internals are considered things like `hass.data` or `entry.runtime_data`, or creating/touching entities. We should mock the library so that it returns what we expect in a real life scenario.

Then we should have HA setup the integration. This way we also test that it tests the right entities are created given the provided source data.

How to test opening a cover? Call the service! And then observe that the calls have been passed on correctly to the library.

Also, for checking the unique id and state. Be sure to look into snapshot tests.

Since your integration is push based, have a look at `niko_home_control`. they got new tests recently and they show how to simulate a push update.

Reply:
I just wonder, in terms of time management, if I tackle the component tests later, since this will require lots of research and learning for me.
For the bronze quality scale I don't need them and this way I can concentrate on getting the other platforms ready, since I want to have a full-functionality cutover for the users.
What is your opinion?

Reply:
I kinda think that tests are really really nice, mostly for an integration that updates differently and has a shitton of devices it supports. 

I'm also happy to add the tests for this platform in the end so you can focus on the rest (and then you add the tests for the rest because you have an example)

Reply:
Yes, they are. I definitely want to have tests. Even this test that does not fit the HA standards showed me two bugs in the code yesterday.
My goal is to bring this integration at least to gold level, so there will be tests!